corediff.go

```diff
@@ -45,25 +60,34 @@ func loadDB(path string) hashDB {
 }
 
 func saveDB(path string, db hashDB) {
-	f, err := os.Create(path)
+	tmpPath := path + ".tmp"
+	f, err := os.Create(tmpPath)
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer f.Close()
+	defer os.Remove(tmpPath)
```

Comment:
Why not use `os.CreateTemp`?

E. g.:

```suggestion
	f, err := os.CreateTemp("", "corediff_temp")
	if err != nil {
		log.Fatal(err)
	}
	defer os.Remove(f.Name())
```

Reply:
great! I'll change it into os.CreateTemp(filepath.Dir(path)) so it will be saved on the same mount and a rename is atomic